### PR TITLE
multi: Convert to use new stdscript package. 

### PR DESCRIPTION
**Pull request by davecgh**
~**This requires ~#2803, #2805, #2807, and #2810**.~

~**NOTE: The various `go.mod` files in the affected modules will need to be updated to use the latest `txscript` module once the aforementioned prerequisite PRs have landed to avoid issues for external consumers, however, this is otherwise ready for review.**~

---

This converts all packages in the repository to make use of the new `stdscript` package for type determination, address extraction, and required signatures extraction instead of the `txscript` methods.  It also removes the aforementioned `txscript` methods once they are no longer used.

Since this involves modifications to a large number of packages in the repository, this PR is split into a series of individual commits such that everything builds and passes all tests (except some temporary unused code linter warnings) each step of the way.  In order to achieve that goal, each commit updates a single package to use the new package.  Finally, all of the unused methods and any associated tests, benchmarks, and examples are then removed one by one.

See each individual commit for more details about each individual change, but the following is a high level overview:

- Converts the following packages to use the new `stdaddr` package:
  - `internal/mining`
  - `internal/mempool`
  - `blockchain/chaingen`
  - `blockchain/indexers`
  - `blockchain`
  - `txscript/sign`
  - `internal/rpcserver`
- Makes minor modifications for consistency and other cleanup while converting
- Removes the following exported funcs, types, and constants:
  - `txscript.ExtractAtomicSwapDataPushes`
  - `txscript.GenerateProvablyPruneableOut`
  - `txscript.MultiSigScript`
  - `txscript.MultisigRedeemScript`
  - `txscript.CalcMultiSigStats`
  - `txscript.IsMultisigScript`
  - `txscript.IsMultisigSigScript`
  - `txscript.ExtractPkScriptAltSigType`
  - `txscript.GetScriptClass`
  - `txscript.GetStakeOutSubclass`
  - `txscript.ExtractPkScriptAddrs`
  - `txscript.MaxDataCarrierSize`
  - `txscript.ScriptClass`
  - `txscript.ErrNotMultisigScript`
  - `txscript.ErrTooManyRequiredSigs`
  - `txscript.ErrTooMuchNullData`

### CODE REVIEW

**Review by matheusd — APPROVED** [review]
Commits tested fine other then the `deadcode` lint errors.

File: blockchain/indexers/existsaddrindex.go [hunk 1]
```diff
@@ -453,47 +445,41 @@ func (idx *ExistsAddrIndex) disconnectBlock(dbTx database.Tx, block, parent *dcr
...
-			// updated to support new script versions.
-			const scriptVersion = 0
-			rs := txscript.MultisigRedeemScriptFromScriptSig(
-				txIn.SignatureScript)
-			class, addrs, _, err := txscript.ExtractPkScriptAddrs(
-				scriptVersion, rs, idx.chain.ChainParams(),
-				isTreasuryEnabled)
+		// Note that the functions used here require v0 scripts.  Hence it
+		// is used for the script version.  This will ultimately need to
+		// updated to support new script versions.
```
Comment (dnldd) [inline blockchain/indexers/existsaddrindex.go:450]: s / ... need to updated to ... / ... need to be updated to ...

Comment (davecgh) [reply blockchain/indexers/existsaddrindex.go]: Added.  Also updated another case with the same typo.

**Review by dnldd — APPROVED** [review]

**Review by rstaudt2 — APPROVED** [review]

**Comment by davecgh** [comment]
I've updated the various `go.mod` files in the affected modules to use the latest `database`, `blockchain/stake, and `txscript` modules to avoid issues for external consumers now that the prerequisite PRs have landed.

**Pull request merged**